### ISSUE TITLE
ETW provider is not discovered automatically for the Stateless ASP.NET Core project type

### ISSUE THREAD

**Issue by nicolave** [Azure/service-fabric-issues#470]
I have created a "Service Fabric Application" using the "Stateless ASP.NET Core" template found in Visual Studio 2017 Pro 15.3.5. I am using using Service Fabric SDK 2.7.198.

And every time I hit F5 to deploy the sample app to my local SF cluster, the Diagnostic Events window notifies me that there are no ETW providers found in the solution. Could this be because it's a .NET core project using the new .csproj format?

Because when I create a "Service Fabric Application" using the "Stateless Service" template, the ETW providers list is automatically updated to include the provider from the sample Stateless1 app. And the stateless's app project is using the old .csproj format.

**Comment by karolz-ms** [comment]
This has been raised on SO as well: https://stackoverflow.com/questions/46361475/etw-provider-is-not-discovered-automatically-for-the-service-fabric-stateless-as 

Tooling version 2.8 should have the fix

**Comment by nicolave** [comment]
I have updated the SF runtime to 6.0.211 and the SF SDK to 2.8.211 using the latest Web Platform Installer. The issue is still present. 
Is there also a new version of the SF tools for Visual Studio 2017 Pro 15.3.5? Because the newly created SF ASP.NET core project still references SF 5.7.198 and 2.7.198 assemblies.

**Comment by karolz-ms** [comment]
OK, sorry for misleading you--turns out the updated Visual Studio tooling has not been released yet. It should be available next week (via Web Platform Installer).

**Comment by nicolave** [comment]
I have upgraded to Visual Studio 15.4 and created both a "Stateless" and a "Stateless ASP.NET Core" sample project. For both projects, no ETW sources were discovered automatically when starting to debug the app (F5). And the Diagnostic Events window is Disconnected.
![no-etw-sources-found](https://user-images.githubusercontent.com/32192827/31538572-ad246e9a-b00e-11e7-8815-b0339616166e.jpg)
Is that the expected behavior? Do I need to manually add ETW sources from now on?

**Comment by karolz-ms** [comment]
No it is not expected. I just tried on my vanilla machine with vanilla VS 15.4 installation and the diagnostics window shows the events from the service.

I suspect a SF tooling setup issue. Please double check that you have Service Fabric tools version 1.8 (Tools | Extensions and Updates | Installed)

This is a shot in the dark, but some other folks have had upgrade problems and were able to eventually pin it down to old version of Azure SDK not being cleaned up properly:
https://developercommunity.visualstudio.com/content/problem/63086/project-template-for-service-fabric-missing-after.html?childToView=128383#comment-128383
It is probably not the same issue that you are suffering from, but the clean up steps in this thread may be helpful.

**Comment by nicolave** [comment]
I have double-checked and I have used SF tools 1.8. It was installed during my upgrade to VS 15.4. So I decided to do a clean install of VS 15.4.

I have removed my VS 15.4 installation and manually uninstalled SF SDK 2.8.211 and SF runtime 6.0.211. I have then installed VS 15.4 from scratch. And then installed SF runtime 6.0.219 and SF SDK 2.8.219 (these were the latest at the time of installation) via the Web Platform Installer. I have double-checked that I have version 1.8.20170927.3 of the SF tools installed in Visual Studio.

I reopened my existing sample projects (both Stateless ASP.NET core and Stateless) and the ETW sources were harvested automatically. I have also created a new sample Stateless ASP.NET core project and the ETW sources were also harvested automatically.

So everything works now. Thank you for your support. 👍
